improver/calibration/rainforest_calibration.py

```diff
@@ -397,6 +398,65 @@ def _make_decreasing(self, probability_data: ndarray) -> ndarray:
         )
         return 0.5 * (upper + lower)
 
+    def _evaluate_probabilities(
+        self,
+        forecast_data: ndarray,
+        input_data: ndarray,
+        forecast_variable: str,
+        forecast_variable_unit: str,
+        output_data: ndarray,
+        model_input_converter: Optional[Callable[[ndarray], object]] = np.array,
+    ):
```

Comment:
Looking back over this, this function changes data in place rather than returning a value. Given all other functions in this class return a value, would it make more sense to be consistent here?

Reply:
Modifying in-place is more efficient for both time and memory, because otherwise we'd need to copy the returned array into `error_probability_cube`

Reply:
Fair enough. Given the bottleneck associated with this part of the process, I'm ok with this approach.